### PR TITLE
Bugfix: Recommend `pytorch-lightning>=2.5.3` for Google Colab notebook

### LINKED ISSUE #2670: [BUG] - TPU training not working in Google Colab
**Describe the bug**
I am aiming to use TPUs to train on Google Colab. 
After update to Google Colab and pytorch xla wheels, the following code no longer runs as I believe TPUs don't run in nodes, they are intrinsically linked to the VMs they are running on. :

**To Reproduce**
```
!pip install cloud-tpu-client==0.10 https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.9-cp37-cp37m-linux_x86_64.whl
!pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 torchtext==0.10.0 -f https://download.pytorch.org/whl/cu111/torch_stable.html
!pip install pyyaml==5.4.1
```

**Expected behavior**
Could alternative code be provided and the documentation updated? I am currently trying to use:

```
!pip install torch~=2.6.0 'torch_xla[tpu]~=2.6.0' \
  -f https://storage.googleapis.com/libtpu-releases/index.html \
  -f https://storage.googleapis.com/libtpu-wheels/index.html

!pip install cloud-tpu-client==0.10
    
# Install the latest PyTorch packages (using CUDA 12.6 builds)
!pip install torch==2.6.0 torchvision==0.21.0 torchtext==0.18 \
    -f https://download.pytorch.org/whl/cu118/torch_stable.html

# Install the latest version of PyYAML
!pip install pyyaml==6.0
```

Whilst this then shows the TPU as available, training is incredibly slow at around 0.18 it/second (when it works) and when running the code [here](https://unit8co.github.io/darts/examples/21-TSMixer-examples.html) , it never gets past cell [11] as it gets stuck here for unknown reasons and no progress bar even shows up. 

**System (please complete the following information):**
 - Python version: [e.g. 3.11.11]
 - darts version [e.g. 0.32.0]

**Additional context**
The code in "To Reproduce" is taken from [here](https://unit8co.github.io/darts/userguide/gpu_and_tpu_usage.html).

### PR DESCRIPTION
**Pull request by daidahao**
Checklist before merging this PR:
- [x] Mentioned all issues that this PR fixes or addresses.
- [x] Summarized the updates of this PR under **Summary**.
- [x] Added an entry under **Unreleased** in the [Changelog](../CHANGELOG.md).

Fixes #2670.

### Summary

Some versions of `pytorch-lightning` may not be comptaible with `torch_vla>=2.7.0` anymore due to deprecation of XLA APIs since 2.7. This was fixed in [Lightning 2.5.3](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.5.3). See Lightning's [#20852](https://github.com/Lightning-AI/pytorch-lightning/issues/20852) for the upstream bug fix.

However, running the "21-TSMixer-examples", the notebook never got past "Model Training". Using the latest nightly build XLA 2.9.0 fixes the issue, as instructed on [pytorch/xla](https://github.com/pytorch/xla). The exact reasons why that is are unknown.

### Other Information

- A working Colab notebook can be accessed here [21-TSMixer-examples.ipynb](https://colab.research.google.com/drive/10lZc3qfgcbNUOQEsytHSVzdJsi6MGnG7?usp=sharing).
- Darts does not currently support `pytorch-lightning>=2.5.3` due to an issue with `lr_find()`, see #2872.

### CODE REVIEW

**Review by dennisbader — COMMENTED** [review]
Thanks for investigation and the the suggested solution @daidahao 🚀 Very nice to see!

As you mentioned, we currently fix pytorch lightning <= 2.5.3 because there was an issue with the `lr_find()` method that didn't give the same learning rate suggestion as in earlier versions. 
I have not yet had the time to raise this issue on their GitHub but will do so soon.

With your update, users wouldn't be able to install the current `darts` package because pypi would complain about incompatible versions. Or did this work for you / could you test it?

Regardless, as soon as we would relax the lightning cap, and if your suggestion is ested then we can merge this PR :)

**Comment by daidahao** [comment]
> Thanks for investigation and the the suggested solution @daidahao 🚀 Very nice to see!
> 
> As you mentioned, we currently fix pytorch lightning <= 2.5.3 because there was an issue with the `lr_find()` method that didn't give the same learning rate suggestion as in earlier versions. I have not yet had the time to raise this issue on their GitHub but will do so soon.
> 
> With your update, users wouldn't be able to install the current `darts` package because pypi would complain about incompatible versions. Or did this work for you / could you test it?
> 
> Regardless, as soon as we would relax the lightning cap, and if your suggestion is ested then we can merge this PR :)

Hi Dennis, I have tested installing the latest darts and "pytorch-lightning>=2.5.3" at the same time on Colab. As you could see in this example [21-TSMixer-examples.ipynb](https://colab.research.google.com/drive/10lZc3qfgcbNUOQEsytHSVzdJsi6MGnG7?usp=sharing), the requirement from darts 0.37.1 could be overwritten and the `pytorch_lightning-2.5.3` would be installed. 

This is not a perfect solution because of Lightning's new `lr_find()` behaviour, neither is installing a nightly build of XLA. But I have tried installing older versions of XLA and they always caused errors with Lightning calling older XLA APIs like `get_ordinal()` or got stuck in model training.

**Comment by dennisbader** [comment]
Thanks @daidahao . In that case I would suggest that we wait first until they released a stable version and then we can continue with this PR. Is that alright with you?

**Comment by daidahao** [comment]
> Thanks @daidahao . In that case I would suggest that we wait first until they released a stable version and then we can continue with this PR. Is that alright with you?

Yes, that is very reasonable! In the meantime, we could point to this PR as a temporary fix if there is any issue with TPU being raised.

**Comment by dennisbader** [comment]
@daidahao, yes sounds great :) In the meantime I should find some time to raise the lr_find issue to lightning.

**Comment by dennisbader** [comment]
@daidahao I have opened [the issue](https://github.com/Lightning-AI/pytorch-lightning/issues/21141#issue-3372317012) now. I'll keep an eye on it :)

**Comment by dennisbader** [comment]
@daidahao, looks like https://github.com/Lightning-AI/pytorch-lightning/pull/21171 will fix the issue :) I'll keep an eye on it and once everything is in place we can go ahead with this PR

**Comment by daidahao** [comment]
@dennisbader Thanks Dennis!
